library/pkcs12.c

```diff
@@ -186,13 +189,18 @@ static void pkcs12_fill_buffer( unsigned char *data, size_t data_len,
     unsigned char *p = data;
     size_t use_len;
 
-    while( data_len > 0 )
+    if( filler != NULL && fill_len != 0 )
```

Comment:
This doesn't look right: why would `filler==NULL` not be an error case when `fill_len!=0`?

Reply:
This was to catch the case where filler was non-NULL and length was zero.

Reply:
NULL filler and nonzero fill length causes NULL pointer dereference. Valid filler pointer and zero fill length causes infinate looop. I have worked around these for the cases we are talking about, but thought it safest to leave this here.